src/byte_format/deserialize.cr

```diff
@@ -83,6 +83,33 @@ struct Crystalizer::ByteFormat
     end
   end
 
+  # :ditto:
+  def deserialize(to type : String.class, size : Range(Int32?, Int32?))
+    string = if max_size = size.end
+               # An alternative to reading limit `max_size + 1` would be to read `max_size` or `max_size - 1`
+               # (depending on `size.excludes_end?`), then peek the next char for `@string_delimiter`, and
+               # then consume it (if it is a valid delimiter) or set a flag for string being out of bounds.
+               @io.gets(@string_delimiter.as(Char), max_size + 1, true) || ""
+             else
+               deserialize type
+             end
+
+    if max_size && ((excludes_end = size.excludes_end?) ? string.size >= max_size : string.size > max_size)
+      raise Error.new "String size not in range: #{size}"
+    end
+
+    if (min_size = size.begin) && string.size < min_size
+      raise Error.new "String size not in range: #{size}"
+    end
+
+    string
+  end
+
+  # Deserializes a `String` from reading from the `io`. String is exactly `size` bytes with no trailing `@string_delimiter`.
+  def deserialize(to type : String.class, size : Int)
+    @io.read_string size
```

Comment:
#13 is already implemented here, no?

Reply:
I don't know, I have doubts to support this natively.
I think it would be better to make serialization/deserialization configurable for this edge cases, like in the stdlib with from_json/to_json methods that can be implemented.

Then inside the type, `deserizalize` can be used directly.

Reply:
What I think, which could be included in this library, is a type which provides this functionality, instead of an annotation.

What we need is an interface, i.e. `abstract struct` - that's essentially it.